app/admin_delete_record.py

```diff
@@ -31,10 +31,22 @@ class Handler(BaseHandler):
     admin_required = True
 
     def get(self):
+        xsrf_tool = XsrfTool()
```

Comment:
No concrete idea, but I wonder if it's possible to share this logic between different admin pages instead of repeating it in all pages (in a common base class of admin pages?). It would also make sure that the logic is also applied to any new admin pages in the future. Or at least report an error when the page does not apply XSRF tool.

Reply:
Yeah, I agree that'd be useful. It might make sense to do it as part of the migration off webapp2 (which I just realized a couple days ago doesn't support Python 3), since that'll involve a lot of changes/refactoring of the handlers anyway.

Reply:
Sounds good.